fahrschein-opentelemetry/src/main/java/org/zalando/fahrschein/opentelemetry/InstrumentedNakadiPublisher.java

```diff
@@ -0,0 +1,56 @@
+package org.zalando.fahrschein.opentelemetry;
+
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.api.common.AttributesBuilder;
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.api.trace.SpanKind;
+import io.opentelemetry.api.trace.StatusCode;
+import io.opentelemetry.api.trace.Tracer;
+import io.opentelemetry.context.Context;
+import org.zalando.fahrschein.NakadiPublisher;
+
+import java.io.IOException;
+import java.util.List;
+
+import static io.opentelemetry.api.trace.SpanKind.PRODUCER;
+
+public final class InstrumentedNakadiPublisher {
+
+    private final NakadiPublisher client;
+    private final Tracer tracer;
+
+    public InstrumentedNakadiPublisher(NakadiPublisher client, Tracer tracer) {
+        this.client = client;
+        this.tracer = tracer;
+    }
+
+    public <T> void publish(String eventName, List<T> events, Span parentSpan) throws IOException {
+        Span childSpan = tracer
+                .spanBuilder("send_" + eventName)
+                .setParent(Context.current().with(parentSpan))
+                .setSpanKind(PRODUCER)
+                .setAttribute("messaging.destination_kind", "topic")
+                .setAttribute("messaging.destination", eventName)
+                .setAttribute("messaging.system", "Nakadi")
+                .setAttribute("messaging.message_payload_size", sizeBucket(events.size()))
+                .startSpan();
+        try {
+            client.publish(eventName, events);
+        } catch (Throwable t) {
+            childSpan.setStatus(StatusCode.ERROR);
+            childSpan.recordException(t);
```

Comment:
Open question: add equivalent exception handling in the opentracing impl? One could argue that we're re-throwing the exception and expecting client logic to record exception details...